src/deadline/job_attachments/asset_sync.py

```diff
@@ -225,7 +225,15 @@ def aggregate_asset_root_manifests(
                     local_root = str(session_dir.joinpath(dir_name))
                     grouped_manifests_by_root[local_root].extend(manifests)
 
-        return grouped_manifests_by_root
+        # Merge the manifests in each root into a single manifest
+        merged_manifests_by_root: dict[str, BaseAssetManifest] = dict()
+        for root, manifests in grouped_manifests_by_root.items():
+            merged_manifest = merge_asset_manifests(manifests)
+
+            if merged_manifest:
```

Comment:
Under what condition is this not true? If it's when `manifests` is an empty list, I think it's more clear to put the if higher up:

```
if len(manifests) > 0:
    merged_manifests_by_root[root] = merge_asset_manifests(manifests)
```

Reply:
Seems to have come from the `asset_sync` code at the bottom of `upload.py` which is now deleted.

Reply:
Yes it's moving existing code around for refactoring. `merge_asset_manifests` returns `BaseAssetManifest | None`, that's why there is a check before adding to `dict[str, BaseAssetManifest]`. Even checks length at a higher level, it still needs a check for the merged manifest.